httplib.h

```diff
@@ -213,6 +213,12 @@ using socket_t = int;
 #include <thread>
 
 #ifdef CPPHTTPLIB_OPENSSL_SUPPORT
+// these are defined in wincrypt.h and it breaks compilation if BoringSSL is used
```

Comment:
Should it be done only for `_WIN32` since it relates to the WinSock2 `wincrypt.h` header file?

Reply:
Sure, good call!